src/org/sosy_lab/common/log/LogManager.java

```diff
@@ -94,15 +93,42 @@
    * <p>Use this method in cases where an expected exception with a useful error message is thrown,
    * e.g. an InvalidConfigurationException.
    *
+   * <p>The message is constructed lazily by concatenating the parts with " ". The caller should not
+   * use string concatenation to create the message in order to increase performance if the message
+   * is never logged. To make individual arguments lazy, use {@link
+   * MoreStrings#lazyString(Supplier)}
+   *
+   * <p>If you want to log an IOException because of a write error, it is recommended to write the
+   * message like "Could not write FOO to file". The final message will then be "Could not write FOO
+   * to file FOO.txt (REASON)".
+   *
+   * @param priority the log level for the message
+   * @param e the occurred exception
+   * @param args the parts of the message (can be an arbitrary number of objects whose {@link
+   *     Object#toString()} method is called)
+   */
+  void logUserException(Level priority, Throwable e, Object... args);
+
+  /**
+   * Log a message by printing its message to the user. The details (e.g., stack trace) are hidden
+   * from the user and logged with a lower log level.
+   *
+   * <p>Use this method in cases where an expected exception with a useful error message is thrown,
+   * e.g. an InvalidConfigurationException.
+   *
+   * <p>The message is constructed lazily from <code>String.format(format, args)</code>. To make
+   * individual arguments lazy, use {@link MoreStrings#lazyString(Supplier)}.
+   *
    * <p>If you want to log an IOException because of a write error, it is recommended to write the
    * message like "Could not write FOO to file". The final message will then be "Could not write FOO
    * to file FOO.txt (REASON)".
    *
    * @param priority the log level for the message
    * @param e the occurred exception
-   * @param additionalMessage an optional message
+   * @param format The format string.
+   * @param args The arguments for the format string.
    */
-  void logUserException(Level priority, Throwable e, @Nullable String additionalMessage);
+  void logfUserException(Level priority, Throwable e, String format, Object... args);
```

Comment:
Removing the old method breaks binary compatibility. We should therefore only add new methods, and remove none of the old ones.

Reply:
Would you prefer to extend `logUserException(Level priority, Throwable e, Object... args)` with the missing param.
> ´logUserException(Level priority, Throwable e, @Nullable String, Object... args)´
Or simply reintroduce the old method

Reply:
The old method needs to be added exactly as it was, otherwise it is not backwards compatible.

Reply:
Done.